flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractJobReconciler.java

```diff
@@ -360,6 +367,25 @@ public boolean reconcileOtherChanges(FlinkResourceContext<CR> ctx) throws Except
         }
     }
 
+    /**
+     * Returns a supplier of all relevant FlinkStateSnapshot resources for a given Flink resource.
+     * If FlinkStateSnapshot resources are disabled, the supplier returns an empty set.
+     *
+     * @param ctx resource context
+     * @return supplier for FlinkStateSnapshot resources
+     */
+    private Supplier<Set<FlinkStateSnapshot>> getFlinkStateSnapshotsSupplier(
+            FlinkResourceContext<CR> ctx) {
+        return () -> {
+            if (FlinkStateSnapshotUtils.isSnapshotResourceEnabled(
+                    ctx.getOperatorConfig(), ctx.getObserveConfig())) {
+                return ctx.getJosdkContext().getSecondaryResources(FlinkStateSnapshot.class);
+            } else {
+                return Set.of();
+            }
+        };
+    }
```

Comment:
Could this logic live in a static utility somewhere else? Maybe even in the timestamp storage or some other utility class